sql/core/src/test/scala/org/apache/spark/sql/PostgreSQLDialectQuerySuite.scala

```diff
@@ -36,7 +36,8 @@ class PostgreSQLDialectQuerySuite extends QueryTest with SharedSparkSession {
     }
 
     Seq("o", "abc", "").foreach { input =>
-      checkAnswer(sql(s"select cast('$input' as boolean)"), Row(null))
+      intercept[IllegalArgumentException](
+        checkAnswer(sql(s"select cast('$input' as boolean)"), Row(null)))
```

Comment:
since we expect error here, we can just run `sql(s"select cast('$input' as boolean)").collect`